instrumentation/logback/logback-appender-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/appender/v1_0/internal/LoggingEventMapper.java

```diff
@@ -278,25 +297,103 @@ private static void captureKeyValuePairAttributes(
     List<KeyValuePair> keyValuePairs = loggingEvent.getKeyValuePairs();
     if (keyValuePairs != null) {
       for (KeyValuePair keyValuePair : keyValuePairs) {
-        Object value = keyValuePair.value;
-        if (value != null) {
-          String key = keyValuePair.key;
-          // preserve type for boolean and numeric values, everything else is converted to String
-          if (value instanceof Boolean) {
-            attributes.put(key, (Boolean) value);
-          } else if (value instanceof Byte
-              || value instanceof Integer
-              || value instanceof Long
-              || value instanceof Short) {
-            attributes.put(key, ((Number) value).longValue());
-          } else if (value instanceof Double || value instanceof Float) {
-            attributes.put(key, ((Number) value).doubleValue());
-          } else {
-            attributes.put(getAttributeKey(key), value.toString());
-          }
+        captureKeyValueAttribute(attributes, keyValuePair.key, keyValuePair.value);
+      }
+    }
+  }
+
+  private static void captureKeyValueAttribute(
```

Comment:
since this method isn't only used for key value attributes any more maybe it should be renamed